src/plugins/data/server/search/search_service.ts

```diff
@@ -61,6 +67,12 @@ import {
 import { aggShardDelay } from '../../common/search/aggs/buckets/shard_delay_fn';
 import { ConfigSchema } from '../../config';
 
+declare module 'src/core/server' {
```

Comment:
it looks unusual =) Are you sure that it's a best option?

Reply:
I think it's the only way today to add things to the `RequestHandlerContext` typescript. I'm also not a big fan.

@elastic/kibana-platform is this the only way?

Reply:
Yes, I verified with platform this is the appropriate way.

Reply:
Yea, until we progress on https://github.com/elastic/kibana/issues/57588, this is the only and correct way.